app/src/main/java/com/github/sdp/mediato/data/Database.java

```diff
@@ -214,4 +221,45 @@ public static CompletableFuture<String> addCollection(String username, Collectio
         return future;
     }
 
+    /**
+     * Method to follow a user
+     *
+     * @param myUsername       the current user's username
+     * @param usernameToFollow the username of the user to follow
+     */
+    public static void followUser(String myUsername, String usernameToFollow) {
+        CompletableFuture<String> future = new CompletableFuture<>();
+        database.getReference()
+                .child(USER_PATH + myUsername + FOLLOWING_PATH + usernameToFollow).setValue(true)
+                .addOnCompleteListener(task -> {
+                    System.out.println(myUsername + " is now following " + usernameToFollow);
+                });
+        database.getReference()
+                .child(USER_PATH + usernameToFollow + FOLLOWERS_PATH + myUsername).setValue(true)
+                .addOnCompleteListener(
+                        task -> {
+                            System.out.println(usernameToFollow + " is now followed by " + myUsername);
+                        });
+    }
+
+    /**
+     * Method to unfollow a user
+     *
+     * @param myUsername       the current user's username
+     * @param usernameToFollow the username of the user to unfollow
+     */
+    public static void unfollowUser(String myUsername, String usernameToFollow) {
+        CompletableFuture<String> future = new CompletableFuture<>();
+        database.getReference()
+                .child(USER_PATH + myUsername + FOLLOWING_PATH + usernameToFollow).setValue(false)
+                .addOnCompleteListener(task -> {
+                    System.out.println(myUsername + " unfollowed " + usernameToFollow);
+                });
+        database.getReference()
+                .child(USER_PATH + usernameToFollow + FOLLOWERS_PATH + myUsername).setValue(false)
+                .addOnCompleteListener(
+                        task -> {
+                            System.out.println(usernameToFollow + " was unfollowed by " + myUsername);
+                        });

```

Comment:
Similar remark, one could event use the proposed method for all 4 pieces of code